astropy/utils/report_info.py

```diff
@@ -0,0 +1,98 @@
+r"""
+A helper script to automate system info in bug reports.
+
+This can be run in two fashions either by downloading the latest version of this
+script and running it on the fly
+```
+curl --silent https://raw.githubusercontent.com/astropy/astropy/main/astropy/utils/report_info.py | python
+```
+
+or, from a local install
+```
+python -c "from astropy.utils.report_info import main; main()"
+```
+
+Running this script should not require anything else than the standard library.
+"""
+
+import platform
+from importlib.metadata import version
+from importlib.util import find_spec
+from itertools import chain
+
+
+def header(name):
+    return [name, "-" * len(name)]
+
+
+def report_platform():
+    lines = [
+        f"{platform.platform() = }",
+        f"{platform.version() = }",
+        f"{platform.python_version() = }",
+    ]
+
+    return lines
+
+
+def report_packages():
+    pkgs = [
+        "astropy",
+        "numpy",
+        "scipy",
+        "matplotlib",
+        "pandas",
+    ]
+    lines = []
+    for p in pkgs:
+        if find_spec(p) is None:
+            v = "--"
+        else:
+            v = version(p)
+        lines.append(f"{p:<20} {v}")
+
+    # pyerfa may not export its package metadata
+    try:
+        import erfa
+    except ImportError:
+        v = "--"
+    else:
+        v = erfa.__version__
+    p = "pyerfa"
+    lines.append(f"{p:<20} {v}")
+    return lines
+
+
+def main():
+    """
+    Examples
+    --------
+    >>> main()  # doctest: +ELLIPSIS
+    platform
+    --------
+    platform.platform() = ...
+    platform.version() = ...
+    platform.python_version() = ...
+
+    packages
+    --------
+    astropy              ...
+    numpy                ...
+    scipy                ...
+    matplotlib           ...
+    pandas               ...
+    pyerfa               ...
+    """
+    report_lines = chain(
```

Comment:
I'm not convinced importing `itertools` just to use `chain()` here once is worth it.

Reply:
I'm confused, what does it cost ? Last I checked `itertools` wasn't an expensive import, and it seems to be imported so often by other parts of the standard library that trying to avoid the overhead seemed futile.

Reply:
`chain()` is not an overly expensive function, but in this case it is very simple to avoid importing `itertools` entirely:
```suggestion
    report_lines = "\n".join([*header("platform"), *report_platform(), "", *header("packages"), *report_packages()])
```
or if you move the `header()` calls inside the `report_*` functions it would be as simple as
```suggestion
    report_lines = "\n".join([*report_platform(), "", *report_packages()])
```

Reply:
I've dropped itertools using unpacking. Thanks !